src/common.rs

```diff
@@ -981,9 +982,87 @@ pub fn get_current_pid() -> Result<Pid, &'static str> {
     inner()
 }
 
+/// MAC address for network interface
+#[derive(PartialEq, Eq, Copy, Clone)]
+pub struct MacAddr {
+    data: [u8; 6],
+}
+
+impl MacAddr {
+    pub const UNSPECIFIED: Self = MacAddr::new();
+
+    #[inline]
+    pub(crate) const fn new() -> Self {
+        Self { data: [0; 6] }
+    }
+
+    pub fn data(&self) -> &[u8; 6] {
+        &self.data
+    }
+}
+
+impl From<[u8; 6]> for MacAddr {
+    fn from(data: [u8; 6]) -> Self {
+        Self { data }
+    }
+}
+
+impl FromStr for MacAddr {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut data = [0; 6];
+        for (index, result) in s
+            .split(':')
+            .take(6)
+            .map(|x| u8::from_str_radix(x, 16))
+            .enumerate()
+        {
+            match result {
+                Ok(result) => {
+                    data[index] = result;
+                }
+                Err(_) => {
+                    break;
+                }
+            }
+            if index == 5 {
```

Comment:
I think it'd be better to ensure that there are not more than 6 numbers. If there are more, then it's invalid and we shouldn't return a `MacAddr` for it. So two things:
1. Please add a test for invalid cases like `10:10:10:10:10:10:10` (with 7 parts).
2. Don't use `take(6)` and don't stop at `index == 5`. Instead, if `index > 5`, you return an error.

Reply:
My bad. The impl of FromStr for MacAddr was no longer used. It was removed